libraries/AP_LeakDetector/AP_LeakDetector.cpp

```diff
@@ -12,12 +12,20 @@ const AP_Param::GroupInfo AP_LeakDetector::var_info[] = {
     // @RebootRequired: True
     AP_GROUPINFO("1_PIN", 1, AP_LeakDetector, _pin[0], -1),
 
+    // @Param: 1_SIGNAL
+    // @DisplayName: Pin mode (analog/digital)
+    // @Description: Signal mode of the pin that connects leak detector 1
+    // @Values: -1:Disabled,0:Analog,1:Digital
+    // @User: Standard
+    // @RebootRequired: True
+    AP_GROUPINFO("1_SIGNAL", 2, AP_LeakDetector, _type[0], -1),
+
     // @Param: 1_LOGIC
     // @DisplayName: Default reading of leak detector when dry
     // @Description: Default reading of leak detector when dry
     // @Values: 0:Low,1:High
     // @User: Standard
-    AP_GROUPINFO("1_LOGIC", 2, AP_LeakDetector, _default_reading[0], 0),
+    AP_GROUPINFO("1_LOGIC", 3, AP_LeakDetector, _default_reading[0], 0),
```

Comment:
I think these parameter id changes can cause issues when upgrading from an older firmware.
Can we keep the old ones as they are and just use new numbers for the new ones?